sqlglot/generator.py

```diff
@@ -3622,7 +3624,7 @@ def userdefinedfunction_sql(self, expression: exp.UserDefinedFunction) -> str:
         expressions = (
             self.wrap(expressions) if expression.args.get("wrapped") else f" {expressions}"
         )
-        return f"{this}{expressions}"
+        return f"{this}{expressions}" if expressions.strip() != "" else this
```

Comment:
Why do we need `strip()` here? Should we instead modify some of the previous lines to make sure `expressions` is an empty string or something? (That's a common pattern in the `Generator` methods.)

Reply:
In the edge case that a procedure had no parameters but did have procedure options, like:
```
CREATE PROCEDURE foo WITH EXECUTE AS OWNER AS BEGIN SELECT 1
```
When I parsed the sql and printed it again from the AST, there was an extra whitespace character in between the procedure name and the WITH keyword like so:
```
CREATE PROCEDURE foo  WITH EXECUTE AS OWNER AS BEGIN SELECT 1
```
Since the `expressions` variable inside the `userdefinedfunction_sql` contains the parameters of the procedure, when there are no parameters, the variable ended up being just a whitespace, so that is why I added `strip()`. I could look a little bit more into it and find a better fix.

Reply:
That's fine, thanks for providing this context 👍